pkg/util/globalerror/grpc.go

```diff
@@ -13,27 +13,24 @@ import (
 	"github.com/grafana/mimir/pkg/mimirpb"
 )
 
-var (
-	grpcClientConnectionIsClosingErr = "grpc: the client connection is closing"
-)
-
 // WrapGRPCErrorWithContextError checks if the given error is a gRPC error corresponding
 // to a standard golang context error, and if it is, wraps the former with the latter.
 // If the given error isn't a gRPC error, or it doesn't correspond to a standard golang
 // context error, the original error is returned.
-func WrapGRPCErrorWithContextError(err error) error {
+func WrapGRPCErrorWithContextError(ctx context.Context, err error) error {
 	if err == nil {
 		return nil
 	}
+	if ctx.Err() == nil {
+		return err
```

Comment:
Why `err` and not `ctx.Err()`?

Reply:
Because at this point there is an error, `err`, it is not nil, but there is no error in the given context `ctx`, so we shouldn't wrap `err` with any context-related errors, but return `err` itself.

Reply:
Oh right, I misread the condition.